### PR TITLE
[SPARK-20249][ML][PYSPARK] Add training summary for LinearSVCModel

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add training summary for LinearSVCModel......

### Why are the changes needed?
 so that user can get the training process status, such as loss value of each iteration and total iteration number.

### Does this PR introduce _any_ user-facing change?
Yes
```LinearSVCModel.summary```
```LinearSVCModel.evaluate```

### How was this patch tested?
new tests

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala [hunk 1]
```diff
@@ -368,6 +388,49 @@ class LinearSVCModel private[classification] (
...
+  @Since("3.1.0")
+  override def summary: LinearSVCTrainingSummary = super.summary
+
+  /**
+   * If the rawPrediction and prediction columns are set, this method returns the current model,
+   * otherwise it generates new columns for them and sets them as columns on a new copy of
+   * the current model
+   */
+  private[classification] def findSummaryModel():
+  (LinearSVCModel, String, String) = {
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala:404]: nit:
`  private[classification] def findSummaryModel(): (LinearSVCModel, String, String) = {` ?

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala [hunk 2]
```diff
@@ -439,3 +502,55 @@ object LinearSVCModel extends MLReadable[LinearSVCModel] {
     }
   }
 }
+
+/**
+ * Abstraction for LinearSVC results for a given model.
+ */
+sealed trait LinearSVCSummary extends BinaryClassificationSummary {
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala:509]: nit: uncessary braces

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala [hunk 3]
```diff
@@ -439,3 +502,55 @@ object LinearSVCModel extends MLReadable[LinearSVCModel] {
...
+}
+
+/**
+ * Abstraction for LinearSVC training results.
+ */
+sealed trait LinearSVCTrainingSummary extends LinearSVCSummary with TrainingSummary {
+}
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala:516]: ditto

File: mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala [hunk 4]
```diff
@@ -267,7 +267,27 @@ class LinearSVC @Since("2.2.0") (
       if (featuresStd(i) != 0.0) rawCoefficients(i) / featuresStd(i) else 0.0
     }
     val intercept = if ($(fitIntercept)) rawCoefficients.last else 0.0
-    copyValues(new LinearSVCModel(uid, Vectors.dense(coefficientArray), intercept))
+    val vec = Vectors.dense(coefficientArray)
+    return createModel(dataset, vec, intercept, objectiveHistory)
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala:271]: I'd like: `createModel(dataset, Vectors.dense(coefficientArray), intercept, objectiveHistory)`
var `vec` and `return` are unnecessary

**Review by zhengruifeng — COMMENTED** [review]
In general LGTM
this is the first case to use rawPrediction to compute aur/prc in summaries.

**Comment by huaxingao** [comment]
cc @srowen

**Comment by srowen** [comment]
Looking good. There's really no downside here right? just works like the other summaries, and is adding this for consistency.

**Comment by huaxingao** [comment]
Right, it works like the other summaries. I don't see any downside for this.

**Comment by huaxingao** [comment]
Thanks! @srowen @zhengruifeng

**Pull request closed without merging**